GitVersionCore/GitPreparer.cs

```diff
@@ -202,10 +202,11 @@ private static Reference GetLocalReference(Repository repository, string branchN
             return repository.Refs.FirstOrDefault(localRef => string.Equals(localRef.CanonicalName, targetBranchName));
         }
 
-        private static DirectReference GetRemoteReference(Repository repository, string branchName, string repositoryUrl)
+        private static DirectReference GetRemoteReference(Repository repository, string branchName, Authentication authentication)
         {
             var targetBranchName = branchName.GetCanonicalBranchName();
-            var remoteReferences = repository.Network.ListReferences(repositoryUrl);
+            var remote = repository.Network.Remotes.Single();
```

Comment:
I think it depends on the `NoFetch` argument. If `NoFetch` isn't set (default) then `GitHelper.NormalizeGitDirectory` calls `GitHelper.EnsureOnlyOneRemoteIsDefined` which throws when the number of remotes != 1.

It's only called from within `GitPreparer.CreateDynamicRepository` which is normally only run on the build server but theoretically could run on a developer's machine.

OK, i will update the PR to create a temp remote, thanks for the advice.